src/Compiler/Microsoft.AspNetCore.Razor.Language/test/Syntax/FindTokenTests.cs

```diff
@@ -0,0 +1,883 @@
+
+        AssertEx.Equal("""RazorCommentLiteral;[ Comment ];""", SyntaxSerializer.Serialize(token).Trim());
+    }
+
+    [Fact]
+    public void RazorComment_05()
+    {
+        var text = """
+        @* Comment$$ *@
+        <div />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.FindToken(position);
+
+        AssertEx.Equal("""RazorCommentLiteral;[ Comment ];""", SyntaxSerializer.Serialize(token).Trim());
+    }
+
+    [Fact]
+    public void RazorComment_06()
+    {
+        var text = """
+        @* Comment $$*@
+        <div />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.FindToken(position);
+
+        AssertEx.Equal("""RazorCommentStar;[*];""", SyntaxSerializer.Serialize(token).Trim());
+    }
+
+    [Fact]
+    public void RazorComment_07()
+    {
+        var text = """
+        @* Comment *$$@
+        <div />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.FindToken(position);
+
+        AssertEx.Equal("""RazorCommentTransition;[@];""", SyntaxSerializer.Serialize(token).Trim());
+    }
+
+    [Fact]
+    public void RazorComment_08()
+    {
+        var text = """
+        @* Comment *@$$
+        <div />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.FindToken(position);
+
+        AssertEx.Equal("""RazorCommentTransition;[@];""", SyntaxSerializer.Serialize(token).Trim());
+    }
+
+    [Fact]
+    public void OutOfRange_NoRangeInDocument()
+    {
+        var text = """
+        <div />$$
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        Assert.ThrowsAny<ArgumentOutOfRangeException>(() => tree.Root.FindToken(-1));
+        Assert.ThrowsAny<ArgumentOutOfRangeException>(() => tree.Root.FindToken(position + 1));
+    }
+
+    [Fact]
+    public void OutOfRange_NoRangeInWhitespaceNode_NoNewline()
+    {
+        var text = """
+        <div $$ />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.DescendantNodes().OfType<SyntaxToken>().Single(t => t.Kind == SyntaxKind.Whitespace);
+        Assert.ThrowsAny<ArgumentOutOfRangeException>(() => token.FindToken(position, includeWhitespace: false));
+        Assert.Same(token, token.FindToken(position, includeWhitespace: true));
+    }
+
+    [Fact]
+    public void OutOfRange_NoRangeInWhitespaceNode_AfterNewline()
+    {
+        var text = """
+        <div
+         $$ />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.DescendantNodes().OfType<SyntaxToken>().Last(t => t.Kind == SyntaxKind.Whitespace);
+        Assert.ThrowsAny<ArgumentOutOfRangeException>(() => token.FindToken(position, includeWhitespace: false));
+        Assert.Same(token, token.FindToken(position, includeWhitespace: true));
+    }
```

Comment:
I suspect its impossible for them to still be present, but if for no other reason than to add more lines to the file, it would be awesome to go through all of the code snippts in https://github.com/dotnet/razor/issues/7630 and https://github.com/dotnet/razor/issues/7505 and include tests for each of the quirks that are called out, and I guess link them to the issues.

Reply:
Didn't quite get it over 1000 lines :(.